powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/AuditingServiceBehavior.java

```diff
@@ -125,4 +129,33 @@ void logSignatureAuditRecord(ActivationRecordEntity activation, SignatureType si
         signatureAuditRepository.save(signatureAuditRecord);
     }
 
+    private String keyValueMapToString(KeyValueMap keyValueMap) {
```

Comment:
This probably should go in a separate `KeyValueMapConverter` class, with methods `fromString(String s)` and `toString()`.

Reply:
Sure. I was thinking whether using JSON may be a better format for storing the data in the DB, what do you think? We would not need a converter.

Reply:
I think we should use JSON (seems better, since DB engines can use their JSON processing features), but I still think that separate converter is better than in place usage of generic `ObjectMapper`.

Reply:
Ok, I'll switch to JSON and move conversion methods it to another class.